### PR TITLE
Add view-layer formatting helpers for templates

### PR DESCRIPTION
**Pull request by madmax983**
## Summary

Introduces a new `format` module with pure, allocation-light helpers for rendering scalar values in Maud templates. These helpers return HTML-escaped `Markup` so output is safe by construction, eliminating hand-rolled `format!` calls in views.

## Key Changes

- **New `autumn/src/format.rs` module** with formatting helpers:
  - `number_to_currency()` / `CurrencyOptions` — format `Decimal` as currency with configurable symbol, precision, and separators
  - `number_with_delimiter()` — render integers/decimals with grouped thousands
  - `pluralize()` / `pluralize_with()` — render count + word with English pluralization rules
  - `truncate()` / `truncate_words()` — shorten text safely without splitting UTF-8 characters
  - `time_ago_in_words()` / `format_datetime()` — render relative and absolute timestamps

- **Shared pluralization logic** between CLI and view layer:
  - Extracted `pluralize_word()` from `autumn-cli/src/generate/naming.rs` into `autumn_web::format`
  - CLI now calls the shared implementation to ensure generated identifiers and rendered text agree on pluralization rules

- **Refactored relative time formatting**:
  - Extracted `relative_time_words()` helper in `format` module
  - `time_zone::time_ago()` now delegates to this shared implementation, eliminating duplication

- **Updated prelude** (`autumn/src/prelude.rs`):
  - Exported all formatting helpers and `Decimal` type behind the `maud` feature flag
  - No additional imports needed beyond the standard prelude

- **Added documentation**:
  - Comprehensive module-level docs with quick example
  - New guide page `docs/guide/format-helpers.md` with usage patterns and copy-paste examples
  - All functions include doc comments with examples

- **Added `rust_decimal` dependency** to workspace and autumn crate

## Implementation Details

- All helpers are pure functions returning `maud::Markup` for safe HTML interpolation
- Currency formatting uses `Decimal::round_dp_with_strategy()` for exact precision without float rounding
- Pluralization covers common English rules (sibilants, consonant+y, irregulars) with escape hatches for edge cases
- Truncation is UTF-8-aware and never splits characters mid-byte
- Relative time uses whole-unit durations (seconds → minutes → hours → days) for readable output
- Comprehensive test coverage for all helpers with edge cases (negative values, zero, multibyte characters, etc.)

https://claude.ai/code/session_01L5dSDFiDeqwxCDiod8DbwB

### CODE REVIEW

File: autumn/src/format.rs [hunk 1]
```diff
@@ -0,0 +1,678 @@
...
+    let len = int_digits.len();
+    let mut out = String::with_capacity(len + len / 3);
+    for (i, ch) in int_digits.chars().enumerate() {
+        if i > 0 && (len - i).is_multiple_of(3) {
+            out.push(separator);
+        }
+        out.push(ch);
+    }
+    out
+}
```
[the remark this answers is not in the document]
Comment (madmax983) [reply autumn/src/format.rs:200]: Applied in b65aa13 — `group_thousands` now counts chars instead of bytes. (Current callers only ever pass ASCII digit strings so the byte/char distinction couldn't actually diverge, but this removes the mismatch defensively as suggested.)

---
_Generated by [Claude Code](https://claude.ai/code)_

File: autumn/src/format.rs [hunk 2]
```diff
@@ -0,0 +1,678 @@
...
+#[cfg(feature = "maud")]
+#[must_use]
+pub fn pluralize(count: i64, singular: &str) -> maud::Markup {
+    let word = if count == 1 {
+        singular.to_owned()
+    } else {
+        pluralize_word(singular)
+    };
+    maud::html! { (count) " " (word) }
+}
```
[the remark this answers is not in the document]
Comment (madmax983) [reply autumn/src/format.rs:217]: Applied in b65aa13 — the singular path now renders `singular` directly with no allocation.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: autumn/src/format.rs [hunk 3]
```diff
@@ -0,0 +1,678 @@
...
+        return format!("{word}es");
+    }
+    if lower.ends_with('y') {
+        let prev = word.chars().rev().nth(1);
+        if prev.is_some_and(|c| !"aeiouAEIOU".contains(c)) {
+            let mut out: String = word.chars().take(word.chars().count() - 1).collect();
+            out.push_str("ies");
+            return out;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (madmax983) [reply autumn/src/format.rs:267]: Applied in b65aa13, with one small change: clippy's `collapsible_if` flagged the nested `if let ... { if ... }`, so I collapsed it to `if let Some(prev) = prefix.chars().next_back() && !"aeiouAEIOU".contains(prev) { ... }`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: autumn/src/format.rs [hunk 4]
```diff
@@ -0,0 +1,678 @@
...
+    if omission_len >= len {
+        // Even the omission marker doesn't fit `len` on its own; clip it so
+        // the total output still honors the caller's length cap.
+        let clipped: String = omission.chars().take(len).collect();
+        return maud::html! { (clipped) };
+    }
+    let keep = len - omission_len;
+    let truncated: String = text.chars().take(keep).collect();
+    maud::html! { (truncated) (omission) }
+}
```
[the remark this answers is not in the document]
Comment (madmax983) [reply autumn/src/format.rs:305]: Applied in b65aa13 — both branches now slice by byte index via `char_indices()` instead of collecting into a new `String`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: autumn/src/format.rs [hunk 5]
```diff
@@ -0,0 +1,678 @@
...
+#[cfg(feature = "maud")]
+#[must_use]
+pub fn truncate_words_with(text: &str, n: usize, omission: &str) -> maud::Markup {
+    let words: Vec<&str> = text.split_whitespace().collect();
+    if words.len() <= n {
+        return maud::html! { (text) };
+    }
+    let truncated = words[..n].join(" ");
+    maud::html! { (truncated) (omission) }
+}
```
[the remark this answers is not in the document]
Comment (madmax983) [reply autumn/src/format.rs:329]: Applied in b65aa13 — renders words via `@for`/`@if` directly instead of allocating a joined `String`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: Cargo.toml [hunk 6]
```diff
@@ -64,6 +64,7 @@ testcontainers-modules = { version = "0.15", features = ["postgres", "redis", "m
 time = { version = ">=0.3, <0.3.48" }
 
 csv = "1.3"
+rust_decimal = "1"
```
[the remark this answers is not in the document]
Comment (madmax983) [reply Cargo.toml:67]: No action needed here — this repo doesn't commit `Cargo.lock` at all; it's listed in `.gitignore` (predates this PR, e.g. still present at the trunk-dev merge base) so there's no lockfile in version control to update. `--locked`/`--frozen`/vendored builds aren't part of this repo's workflow.

---
_Generated by [Claude Code](https://claude.ai/code)_
